assets/js/src/newsletters/types.jsx

```diff
@@ -73,98 +81,96 @@ class NewsletterTypes extends React.Component {
-        action: (function action() {
-          return (
-            <a
-              className="button button-primary"
-              data-automation-id="create_standard"
-              onClick={createStandardNewsletter}
-              role="button"
-              tabIndex={0}
-              onKeyDown={(event) => {
-                if ((['keydown', 'keypress'].includes(event.type) && ['Enter', ' '].includes(event.key))
-                ) {
-                  event.preventDefault();
-                  createStandardNewsletter();
-                }
-              }}
-            >
-              {MailPoet.I18n.t('create')}
-            </a>
-          );
-        }()),
+        action: (
+          <a
+            className="button button-primary"
+            data-automation-id="create_standard"
+            onClick={createStandardNewsletter}
+            role="button"
+            tabIndex={0}
+            disabled={this.state.isCreating}
+            onKeyDown={(event) => {
+              if ((['keydown', 'keypress'].includes(event.type) && ['Enter', ' '].includes(event.key))
+              ) {
+                event.preventDefault();
+                createStandardNewsletter();
+              }
+            }}
+          >
+            {MailPoet.I18n.t('create')}
+          </a>
+        ),
       },
       {
         slug: 'welcome',
         title: MailPoet.I18n.t('welcomeNewsletterTypeTitle'),
         description: MailPoet.I18n.t('welcomeNewsletterTypeDescription'),
         videoGuide: 'https://kb.mailpoet.com/article/254-video-guide-to-welcome-emails',
         videoGuideBeacon: '5b05ebf20428635ba8b2aa53',
-        action: (function action() {
-          return (
-            <a
-              className="button button-primary"
-              onClick={createWelcomeNewsletter}
-              data-automation-id="create_welcome"
-              onKeyDown={(event) => {
-                if ((['keydown', 'keypress'].includes(event.type) && ['Enter', ' '].includes(event.key))
-                ) {
-                  event.preventDefault();
-                  createWelcomeNewsletter();
-                }
-              }}
-              role="button"
-              tabIndex={0}
-            >
-              {MailPoet.I18n.t('setUp')}
-            </a>
-          );
-        }()),
+        action: (
+          <a
+            className="button button-primary"
+            onClick={createWelcomeNewsletter}
+            data-automation-id="create_welcome"
+            disabled={this.state.isCreating}
+            onKeyDown={(event) => {
+              if ((['keydown', 'keypress'].includes(event.type) && ['Enter', ' '].includes(event.key))
+              ) {
+                event.preventDefault();
+                createWelcomeNewsletter();
+              }
+            }}
+            role="button"
+            tabIndex={0}
+          >
+            {MailPoet.I18n.t('setUp')}
+          </a>
+        ),
```

Comment:
I see you unwrapped the closure and then added the `disabled` attribute. It's all in one commit so I don't know why. Then I have to ask - did you unwrap it just because it's nicer or because it's needed?

Reply:
Because it's needed. if I keep the closure/anonymous function, then I can't access the class instance using `this`.